### PR TITLE
B2B-651 - Custom popover view for pagination

### PR DESCRIPTION
**Pull request by chris-atomlearning**
This ticket is part of this B2B ticket:
https://atomlearningltd.atlassian.net/browse/B2B-651
AS A Quest student
I WANT to find the navigation easy and intuitive, even if I am a 5-year-old
SO THAT I can do my best in the assessment without being confused or distracted

The pagination popover is exactly what we needed to display, just with a different trigger, and no horizontal numbered list of pages.

I considered pulling the <PaginationPopover> component out so it could be added directly, but it would need the Provider and ColorScheme around it, and that quickly seemed impractical.

So I have added a couple of new props, to turn the "popover only" view on, and specify what the trigger looks like. 
I'm open to suggestions on naming and other alternatives but this seems like a really neat way to achieve what we want.

image

### CODE REVIEW

File: lib/src/components/pagination/Pagination.tsx [hunk 1]
```diff
@@ -17,11 +18,13 @@ export const Pagination = ({
   disabledPages = [],
   onItemHover = () => null,
   labels = {},
+  showPopoverTrigger = false,
+  popoverTriggerLabel = null,
```
Comment (chris-atomlearning) [inline lib/src/components/pagination/Pagination.tsx:22]: Do these props make sense? Is there a neater way?

Comment (chris-atomlearning) [reply lib/src/components/pagination/Pagination.tsx]: Good shout, I like that, works really nice :)

File: lib/src/components/pagination/Pagination.test.tsx [hunk 2]
```diff
@@ -216,4 +219,25 @@ describe('Pagination component', () => {
       await waitFor(() => axe(container), { timeout: 5000 })
     ).toHaveNoViolations()
   })
+
+  it('renders with custom popover trigger if turned on', async () => {
+    await render(
+      <Pagination pagesCount={10} selectedPage={1}>
+        <Pagination.Popover>
+          <Flex gap={1} align="center" css={{ cursor: 'pointer' }}>
```
Comment (LimeWub) [inline lib/src/components/pagination/Pagination.test.tsx:227]: Is the Flex here needed for `asChild` to work?

**Review by LimeWub — APPROVED** [review]

File: lib/src/components/pagination/Pagination.test.tsx (hunk 2 above)
Comment (chris-atomlearning) [reply lib/src/components/pagination/Pagination.test.tsx:227]: Ah no, in the sandbox version I had an icon as well and I just copied it without the icon for some reason. I'll add the icon back in so it's the same as how we're going to use it.

**Pull request merged**